src/python/pants/reporting/html_reporter.py

```diff
@@ -381,9 +382,9 @@ def _render_message(self, *msg_elements):
       if isinstance(element, string_types):
         element = [element]
 
-      # Map assumes None for missing values, so this will pick the default for those.
-      (text, detail, detail_id, detail_initially_visible) = \
-        map(lambda x, y: x or y, element, ('', None, None, False))
+      # zip_longest assumes None for missing values, so this generator will pick the default for those.
+      default_values = ('', None, None, False)
+      (text, detail, detail_id, detail_initially_visible) = (x or y for x, y in zip_longest(element, default_values))
```

Comment:
This was tricky. In Py2, `map()` defaults to None when zipping like this and one of the iterables in shorter. In Py3, it no longer does that, so we have to use `zip_longest()`.

I tried to make this more readable with the `default_values` variable and converting to a generator. 

I don't think there's a unit test, but every pants goal fails when this piece is broken, so I think it's good to go.

Reply:
Yep, sounds reasonable. Thanks!